Automattic/woocommerce-services
# Show admin notice for missing product dimension errors

Fixes #1386.

Prevents rate requests from dispatching when a product is missing dimension value(s). Surfaces the error in `/wp-admin` in the form of a notice and on the cart page if debugging is enabled.

![screen shot 2018-11-08 at 4 20 43 pm](https://user-images.githubusercontent.com/63922/48233607-2073bc00-e373-11e8-8015-2ebf98202fd8.png)

![screen shot 2018-11-08 at 4 24 10 pm](https://user-images.githubusercontent.com/63922/48233608-249fd980-e373-11e8-85b4-dac0dd3da695.png)

**To test:**
* Add a product to your cart that is either missing a dimension value (length, width, height) or has `0` as the value
* Verify that the fallback rate is used (if configured)
* Verify that the admin notice is displayed on the dashboard
* Verify that entering dimensions for the product dismissed the notice
